### ISSUE TITLE
Commercial Licensing Inquiry for yolov9

### ISSUE THREAD

**Issue by ankit-unifynow** [WongKinYiu/yolov9#649]
Hello @WongKinYiu ,

I am interested in using **YOLOv9 (GPL‑3.0)** in a commercial product without making my proprietary code open source. Under GPL‑3.0, this is not permitted. Would you be open to discussing a separate licensing agreement or commercial license?

Thank you for your time and for your contributions to the community.

Best regards,  
Ankit
Unifynow AI

**Comment by ankit-unifynow** [comment]
Hey @WongKinYiu ,

Any update on this ??

**Comment by ankit-unifynow** [comment]
Hey @WongKinYiu ,

Awaiting for your response

**Comment by jfallz** [comment]
You can always just make it open source, why not?

**Comment by faridht** [comment]
@ankit-unifynow 

How about using this one instead? It's MIT license.

https://github.com/MultimediaTechLab/YOLO

**Comment by XMR13** [comment]
Hey @faridht  i have trouble using that repo to train my model, is it possible if I want to ask you a couple of things? Thanks

**Comment by faridht** [comment]
@XMR13 although I might not always be able to respond quickly, I can surely help you out. I worked with the code on that repo and managed to train some models with it.

**Comment by XMR13** [comment]
@faridht appreciate the help; I want to ask you a couple of questions.

1. How do you actually structure your dataset format (is it COCO or YOLO format)? I'm using both, and when I tried to train, my maps were not improving even after 5/6 epochs.

2. How do you train your model (args, setup, etc.)? I read the docs, and it confuses me.

3. Is there any patch that you did to fix the repo? Because I take a look at the issue and PR menu, it seems like the repo itself is ridden with bugs and incomplete features.

4. Do you manage to export it into other formats (ONNX, TensorRT)?

That's my question. I appreciate you taking your time to respond to my questions. Thanks!

**Comment by faridht** [comment]
@XMR13 

1. I resorted to using COCO format (.json). However, due to some issue with the main code, I had to remove all mentions of segmentation in the annotation json files. The code as-is seems to demand the presence of segmentation data. I never had the chance to patch it, so I just clean up the json files from segmentation references instead.

2. The only deviations that I recall doing were disabling the mosaic augmentation (at least as it was counter-intuitive for my case), but more importantly I reduced the learning rate. That helped enhance the mAPs. My command usually looks like this: 

```
python yolo/lazy.py task=train dataset=custom task.data.batch_size=65 task.data.data_augment.Mosaic=0 model=v9-t name=custom-9t
```

3. I've done some patches locally, but I can't recall all of them unfortunately. One I remember was customizing the model name of tensorRT to reflect input resolution and batch name.
 
4. Yes I managed to do the exports. I'll share code I made based on a lot of digging in "issues" tab. First run `convert_to_pt.py` then `convert_to_onnx_trt.py`. I'll update this comment later to add references, but I don't have the links for them now. 

[convert_to_pt.py](https://github.com/user-attachments/files/22899020/convert_to_pt.py)
[convert_to_onnx_trt.py](https://github.com/user-attachments/files/22899021/convert_to_onnx_trt.py)
